### ISSUE TITLE
fix(remote): index and launch Amiga ADF games

### ISSUE THREAD

**Issue by ransom1122** [wizzomafizzo/mrext#32]
Make Amiga games appear in Remote’s search database and launch through the Minimig core.

- [x] Index `.adf` files from Amiga game folders.
- [x] Launch indexed ADF games through Minimig.
- [x] Support AmigaVision game and demo listings.

HDF launching and selecting alternate Minimig configurations are separate features and are not included in this issue.

**Comment by wizzomafizzo** [comment]
Hi. Yes that's normal unfortunately. I was not able to work out the correct parameters for launching Amiga with an mgl file from the core's source code. If you have an example working mgl file for the amiga I'd be happy to add support, or know someone who knows.

**Comment by ransom1122** [comment]
I dont know what mgl files are unfortunately, but I know the ins and outs of amiga on mister...

Is there any more info on mgl files anywhere or can they be edited with notepad++ ?

Great work with Remote by the way.. nice job 👏

**Comment by wizzomafizzo** [comment]
Thank you :) glad to hear you like it

I wrote an article how they work: https://mister-devel.github.io/MkDocs_MiSTer/advanced/mgl/

They're not that complicated though. It's a file that tells mister what core to launch with what files to load into it. You can see at the bottom of the article how to figure out what the mgl arguments should be. This does not exist on Amiga! I guess because it's a very old core? Not sure.

In the future I'd also like to add launching directly into AmigaVision. Although it hasn't been a priority since AmigaVision is already a very nice experience haha

**Comment by ransom1122** [comment]
This is beyond my knowledge... Is this dependant on wether the core creator has created some code, or can the amiga  core do it as long as you know index/type/path information?

I would assume it should be something like this?

<mistergamedescription>
    <rbf>_Computer/Minimig_20230216</rbf>
    <file delay="1" type="s" index="2" path="hdf/Game.hdf"/>
    <file delay="1" type="s" index="4" path="adf/game.adf"/>
</mistergamedescription>

But thats just me making things up now, with no knowledge

**Comment by wizzomafizzo** [comment]
yes! just like that. you can test it by putting that file somewhere in the menu and launching yourself. i will try it later today and  find some amiga files. i don't use the core much myself tbh

one caveat here is that my understanding is the mgl file relies on the core auto-booting whatever file gets loaded in the index. like when you stick a cart in an nes it will auto-boot it. but for computer cores it will auto-boot the hard drive image but not necessarily the cd image. that's ok though as long as you know how to create an amiga os image that will auto-run the disc image

**Comment by ransom1122** [comment]
There's only two formats the amiga boots. adf or hdf... hope things work out. Good luck

**Comment by wizzomafizzo** [comment]
Hi! Could you please try this test version out for me? I have added basic support for launching .adf files and showing them in the search. You will need to:

1. Copy the remote.sh file in this .zip file to the Scripts folder on your mister, replacing the existing remote.sh file
2. Power cycle your mister (or, in remote, go to Settings -> Remote -> Advanced and press Restart remote service)
3. Go to the Search page and press the Generate index button next to the System dropdown
4. Refresh the page and then check if Amiga is now in the systems list and you can launch a game

If it does not work, please let me know, and I have another copy you can try which I know definitely works for other people.

[remote-0.2.3alpha4.zip](https://github.com/wizzomafizzo/mrext/files/12216349/remote-0.2.3alpha4.zip)

**Comment by ransom1122** [comment]
Platform Amiga is now listed in "Search" Drop down list. It is not listed in "Systems" List (Nor is Commodore 64)

ADF's now boot..
HDF's are not found..

Scans for Amiga can only be found on both Interal SD Card and External HDD..



"The other problem is Minimig Core has two main configurations, one being for the Amiga 500 & the other for the Amiga 1200...

Depending on what Startup Amiga Configuration people's minimig core is setup as, some ADF games wont work, For Example...\

My Minimig Core is setup as Amiga 1200 bootup, which makes booting HDF Games a breeze, but at the same time 
booting older games on ADF may not work as some ADF's are designed to run on A500 Config (Which can be manually changed after minimig core is booted)

For example Bubble Bobble ADF "will boot" but not work with my default Amiga 1200 Bootup config from Remote"



I don't think there's an option to launch with remote and use a secondary amiga config is there?

Can you also add HDF extenstion as it will solve most problems as HDF games will work because they are optimised to work with A1200 default config.

**Comment by wizzomafizzo** [comment]
The original missing-Amiga behavior is fixed by current ADF and AmigaVision indexing and launch support. Broader HDF and Minimig configuration support is not planned as part of this issue.
